servers/visual/visual_server_viewport.cpp

```diff
@@ -178,11 +178,14 @@ void VisualServerViewport::_draw_viewport(Viewport *p_viewport, ARVRInterface::E
 		VSG::rasterizer->restore_render_target();
 
 		if (scenario_draw_canvas_bg && canvas_map.front() && canvas_map.front()->key().layer > scenario_canvas_max_layer) {
+			Ref<ARVRInterface> arvr_interface = ARVRServer::get_singleton()->get_primary_interface();
 
-			if (can_draw_3d) {
-				VSG::scene->render_camera(p_viewport->camera, p_viewport->scenario, p_viewport->size, p_viewport->shadow_atlas);
-			} else {
+			if (!can_draw_3d) {
 				VSG::scene->render_empty_scene(p_viewport->scenario, p_viewport->shadow_atlas);
+			} else if (p_viewport->use_arvr && arvr_interface.is_valid()) {
+				VSG::scene->render_camera(arvr_interface, p_eye, p_viewport->camera, p_viewport->scenario, p_viewport->size, p_viewport->shadow_atlas, p_viewport->keep_3d_linear);
+			} else {
+				VSG::scene->render_camera(p_viewport->camera, p_viewport->scenario, p_viewport->size, p_viewport->shadow_atlas, p_viewport->keep_3d_linear);
```

Comment:
I would put this in the RasterizerStorageGLES3::RenderTarget as a flag instead of passing even more stuff via this function.

Reply:
Done, much cleaner indeed, didn't realize I had access to this structure in the render code but that makes sense off course.